File: Java-To-Python-Conversion/AverageGrades.java
public class AverageGrades {
    static int[] average_grades(int grades[][], int weights[]) {
        if (grades == null || weights == null)
            return new int[0];

        int students = grades.length;
        if (students == 0)
            return new int[0];

        int comps = weights.length;
        int[] result = new int[students];

        for (int i = 0; i < students; i++) {
            int sum = 0;
            for (int j = 0; j < comps; j++) {
                sum += grades[i][j] * weights[j];
            }
            result[i] = sum / 100;
        }

        return result;
    }

    public static void main(String[] args) {
        int[][] grades = { { 51, 83, 28 }, { 0, 38, 95 } };
        int[] weights = { 30, 40, 30 };

        int[] av = average_grades(grades, weights);

        System.out.println(av);
    }

}

File: university_courseworks/year1/cs1ip/coursework1/average_grades.py
def AverageGrades(grades, weights):
    
    # Null / empty check
    if grades is None or weights is None:
        return []

    students = len(grades)
    if students == 0:
        return []

    comps = len(weights)
    result = [0] * students  # Creates a list of length 'students' pre-filled with 0

    for i in range(students):
        total_sum = 0
        
        for j in range(comps):
            total_sum += grades[i][j] * weights[j]
        
        # Integer division in Python is // (or / if you want floating point)
        result[i] = total_sum // 100

    return result


if __name__ == "__main__":

    # Define the input data in main
    grades = [[51, 83, 28], [0, 38, 95]]
    weights = [30, 40, 30]

    av = AverageGrades(grades, weights)
    print(av)  # Outputs: [56, 43]

